### PR TITLE
Generic button shortcode

### PR DESCRIPTION
**Pull request by NatasaKralj**
This PR makes different kinds of buttons available to use for links within content.

When building locally, you can view the different kinds available on the indentation-spacing-test page.

### CODE REVIEW

File: layouts/shortcodes/button.html [hunk 1]
```diff
@@ -0,0 +1,5 @@
+{{ $color := .Get "color" | default "info" }}
+{{ $link := .Get "href" }}
+{{ $text := .Get "text" | default "Download" }}
+{{ $title := .Get "title" }}
+<a class="btn btn-{{ $color }}" href="{{ $link }}" title="{{ $title }}" target="_blank">{{ $text }}</a>
```
Comment (MarkvanMents) [inline layouts/shortcodes/button.html:6]: The only issue here is if someone puts in a color which is not one of the standard ones.
For example "color="#FF00FF"
However, the same issue holds for alerts, and it doesn't crash, so let's assume that people will check that it works, and we can link to a list of the valid ones (or list our preferred ones) from our documentation.

**Review by MarkvanMents — APPROVED** [review]
See comment - perhaps need to make sure we document it well?

File: layouts/shortcodes/button.html (hunk 1 above)
Comment (NatasaKralj) [reply layouts/shortcodes/button.html:6]: I agree, it has to be documented well. I'll add a section in the Use the Docs Site doc to cover all options before merging.
I don't think we should run strict tests on what content is submitted in a button shortcode. As you mentioned, the same is possible with alerts. It should be on the writer to double check if it looks the way they want. Bad links will still fail in htmltest.
As far as default colors and styling go, we can always adjust those later.

Comment (MarkvanMents) [reply layouts/shortcodes/button.html:6]: Yes - good plan. That's why I approved it as well as making the comment.

File: assets/scss/_variables_project.scss [hunk 2]
```diff
@@ -46,6 +46,24 @@ $gray-800: #3B4251 !default; // MX Midnight 80
 $gray-900: #0A1325 !default; // MX Midnight 100 
 $black: #000 !default;
 
+// NK - copied from themes\docsy\assets\vendor\bootstrap\scss\_variables.scss to add extra theme color,
```
Comment (NatasaKralj) [inline assets/scss/_variables_project.scss]: @MarkvanMents I added theme colors into variables, as a way to override the list of default theme colors. This lets us add any color keyword we want.  
This way I just define a new theme color, give it a color code > Docsy/Bootstrap does the rest. It calculates all the color gradients for any action connected to the button. 
This way is cleaner than copying lots of internal Bootstrap styling for buttons, just to define a new button. There are lots of operations done (hover, click, disable effects), that apply color changes based on the initial theme color chosen.

Comment (MarkvanMents) [reply assets/scss/_variables_project.scss]: Not sure I understand how this works. You'll have to explain it sometime. But it seems OK.

**Review by MarkvanMents — APPROVED** [review]
Looks OK - I'd quite like to go through what you achieve in creating $theme-colors, but you don't have to wait before merging.

The only concern I have is with **outline-light**. The text is so light it is difficult to see.
image

Is there any scope to change the text color to black (while we are in light mode - we can look at the impact of dark mode some other time 😀).

File: assets/scss/_variables_project.scss (hunk 2 above)
Comment (NatasaKralj) [reply assets/scss/_variables_project.scss]: I ended up removing the theme color expansion. It's nice to know how to do this, if we ever need to, but we don't have much of a use case for it ATM.

Comment (MarkvanMents) [reply assets/scss/_variables_project.scss]: Great research. I'm sure it will come in useful in doing improvements in other areas of the site.

**Pull request merged**